### PR TITLE
bugfix: set GO111MODULE to on when use docker build

### PR DESCRIPTION
**Pull request by Starnop**
Signed-off-by: Starnop <starnop@163.com>

### Ⅰ. Describe what this PR did
We have updated the go version to 1.12.6 and use go mod to manage the dependency. So we should set the `GO111MODULE` env to `on` when building with docker. Otherwise we would get errors like this：

```
apis/types/df_get_task.go:11:2: cannot find package "github.com/go-openapi/errors" in any of:
	/usr/local/go/src/github.com/go-openapi/errors (from $GOROOT)
	/go/src/github.com/go-openapi/errors (from $GOPATH)
apis/types/df_get_task.go:12:2: cannot find package "github.com/go-openapi/strfmt" in any of:
	/usr/local/go/src/github.com/go-openapi/strfmt (from $GOROOT)
	/go/src/github.com/go-openapi/strfmt (from $GOPATH)
apis/types/df_get_task.go:13:2: cannot find package "github.com/go-openapi/swag" in any of:
	/usr/local/go/src/github.com/go-openapi/swag (from $GOROOT)
	/go/src/github.com/go-openapi/swag (from $GOPATH)
......
```

### Ⅱ. Does this pull request fix one issue?

None.
### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
None.

### Ⅳ. Describe how to verify it
Use command `docker build USE_DOCKER=1` and it can be executed successfully

### CODE REVIEW

**Comment by yeya24** [comment]
lgtm

File: hack/build.sh [hunk 1]
```diff
@@ -59,10 +59,12 @@ build-docker() {
         -e GOOS="${GOOS}"                                                 \
         -e GOARCH="${GOARCH}"                                             \
         -e CGO_ENABLED=1                                                  \
```
Comment (yeya24) [inline hack/build.sh]: Do we need to enable CGO here? In my mac os env it will produce some unrelated errors:
``` bash
net
# net
/usr/local/go/src/net/cgo_bsd.go:15:72: could not determine kind of name for C.AI_MASK
# os/user
/usr/local/go/src/os/user/getgrouplist_darwin.go: In function 'mygetgrouplist':
/usr/local/go/src/os/user/getgrouplist_darwin.go:16:11: warning: implicit declaration of function 'getgrouplist' [-Wimplicit-function-declaration]
  int rv = getgrouplist(user, (int) group, buf, ngroups);
           ^~~~~~~~~~~~
BUILD: dfget in /Users/yeya24/hub/gowork/Dragonfly/bin/dfget
net
# net
/usr/local/go/src/net/cgo_bsd.go:15:72: could not determine kind of name for C.AI_MASK
# os/user
/usr/local/go/src/os/user/getgrouplist_darwin.go: In function 'mygetgrouplist':
/usr/local/go/src/os/user/getgrouplist_darwin.go:16:11: warning: implicit declaration of function 'getgrouplist' [-Wimplicit-function-declaration]
  int rv = getgrouplist(user, (int) group, buf, ngroups);
           ^~~~~~~~~~~~
```

File: hack/build.sh [hunk 2]
```diff
@@ -59,10 +59,12 @@ build-docker() {
...
+        -e GO111MODULE=on                                                 \
+        -e GOPROXY=https://goproxy.io                                     \
         -w /go/src/${PKG}                                                 \
         ${BUILD_IMAGE}                                                    \
         go install -v -pkgdir /go/pkg -ldflags "${LDFLAGS}" ./cmd/"$2"
```
Comment (yeya24) [inline hack/build.sh]: Maybe change this to the cmd below?
``` Go
go build -o "$1" -ldflags "${LDFLAGS}" ./cmd/"$2"
```

**Comment by yeya24** [comment]
lgtm

File: hack/build.sh [hunk 3]
```diff
@@ -52,17 +48,19 @@ build-docker() {
         --rm                                                              \
         -ti                                                               \
         -u "$(id -u)":"$(id -g)"                                          \
-        -v "$(pwd)"/.go:/go                                               \
+        -v "$(pwd)"/.go:/.go                                              \
```
Comment (lowzj) [inline hack/build.sh]: Why did you change `/go` into `/.go`? `/.go` is not in the `GOPATH` in the building container, it will download packages and modules every time when we use `make USE_DOCKER=1 build-client`.
